.github/workflows/pixlet.yml

```diff
@@ -0,0 +1,113 @@
+name: pixlet
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.16.12'
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Install Linux dependencies
+      if: matrix.os == 'ubuntu-latest'
+      run: sudo apt-get install -y libwebp-dev
+
+    - name: Install macOS dependencies
+      if: matrix.os == 'macos-latest'
+      run: brew install webp
+
+    - name: Build
+      run: make build
+
+    - name: Test
+      run: make test
+
+  build-release-artifacts:
+    name: Build Release Artifacts
+    needs: build-and-test
+    if: github.ref == 'refs/heads/main'
```

Comment:
IIUC, this will mean the release artifacts are not built for workflow instances triggered by a tag, so they won't be available for the `create-release` job to download.